File: Data Structure Basics/Java Basic Data Structures/Java Basic Data Structures/src/BasicDataStructures.java
import java.util.*;
public class BasicDataStructures
{
    public static void main(String[] args)
    {
        int[] numbers = {1,2,3}; // initialize array
        System.out.println(numbers.length); // prints 3

        String[] clothes = new String[5]; // initialize array capable of holding 5 strings
        clothes[0] = "hi"; // sets first element

        for (int i = 0; i < numbers.length; i++) // increments by 1
        {
            System.out.println(numbers[i]);
        }

        for (int number: numbers) // for each loops directly fetches elements
        {
            System.out.println(number);
        }
    }
}

class LinkedListNode
{
    public int val;
    public LinkedListNode next;

    public LinkedListNode(int val)
    {
        this.val = val;
    }
}

class TreeNode
{
    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode( int val )
    {
        this.val = val;
    }
}

class NTreeNode
{
    public int val;
    public ArrayList<NTreeNode> children;

    public NTreeNode(int val)
    {
        this.val = val;
        this.children = new ArrayList<NTreeNode>();
    }
}

File: Data Structure Basics/Javascript Basic Data Structures/Basic Data Structures.js
const arr = [0,1,2,3,4,5];
let subarray = arr.slice(2,5); // 2,3,4
subarray = arr.slice(-2); // 4,5

const numbers = [2,6,13,5];
// c-like for loop fetches elements using indices
for (let i = 0; i < numbers.length; i++)
{
    const number = numbers[i];
    console.log(number);
}

// foreach loop directly fetches elements and their indices
numbers.forEach(function (item,index)
{
    console.log(`${item} ${index}`);
    // 2 0
    // 6 1
    // 13 2
    // 5 3
})

// for of loops directly fetches elements

for (const number of numbers)
{
    console.log(number);
}

class LinkedListNode
{
    constructor(val, next = null)
    {
        this.val = val;
        this.next = next;
    }
}

function LinkedListNode(val, next)
{
    this.val = val;
    this.next = next;

    const LinkedListNode = {val: 1, next: {val: 2, next: null}};
}

class TreeNode
{
    constructor(val)
    {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class NTreeNode
{
    constructor(val)
    {
        this.val = val;
        this.children = [];
    }
}